Update for new fix structure

WalterKolczynski-NOAA opened the pull request:
## DESCRIPTION OF CHANGES: 
Global workflow recently updated their fix directory structure to incorporate versioning using dates within the directory tree. Additionally, some files were moved and directories renamed (including removing all the 'fix_' prefixes of subdirectories). This commit updates the fix paths in UFS utils to match the new directory structure, and also take version numbers for each of the fix component directories, similar to how versions are used for modulefiles.)

## TESTS CONDUCTED: 
No new unit tests required.

- [x] cmake tests all pass on Hera for Intel
- [x] cmake tests all pass on Hera for gnu
- [x] cmake tests all pass on Orion
- [x] cmake tests all pass on Cactus
- [x] CI tests all pass
- [x] Global-workflow cycled test on Hera
- [x] Global-workflow coupled test on Hera

## DEPENDENCIES:
None

## DOCUMENTATION:
None needed

## ISSUE: 
Fixes #687
Refs: NOAA-EMC/global-workflow#966

## CONTRIBUTORS (optional): 
@KateFriedman-NOAA did a lot of the work before I finished it off and tested.

WalterKolczynski-NOAA committed:
Update for new fix strucutre

Global workflow recently updated their fix directory structure to incorporate
versioning using dates within the directory tree. Additionally, some files were
moved and directories renamed (including removing all the 'fix_' prefixes of
subdirectories). This commit updates the fix paths in UFS utils to match the
new directory structure, and also take version numbers for each of the fix
component directories, similar to how versions are used for modulefiles.)

Fixes #687
Refs: NOAA-EMC/global-workflow#966

WalterKolczynski-NOAA commented:
Kate didn't open a PR before she left, but I found her stuff and finished it off.

I don't have access to Jet to test there. I do not know who maintains fix on the other platforms not already tested or if they have been updated.

DavidHuber-NOAA commented:
@WalterKolczynski-NOAA S4 has the updated fix directories.  I will run a cycled and S2S test on that system.

I can run the UFS_UTILS regression tests on Jet, but the global workflow port has not been updated since March, so I won't be able to run a cycled or coupled test at this time.  Would you like me to go ahead with the RTs?

DavidHuber-NOAA commented:
@WalterKolczynski-NOAA gdas_init is looking for the mosaic files in the wrong location.  The chgres scripts all set `FIX_ORO=${FIX_FV3}/fix_fv3_gmted2010` instead of `${FIX_FV3}/orog`.

DavidHuber-NOAA commented:
@WalterKolczynski-NOAA OK, that makes sense, thanks.

DavidHuber-NOAA commented:
@WalterKolczynski-NOAA I will run RTs on S4 instead.

GeorgeGayno-NOAA commented:
> @WalterKolczynski-NOAA gdas_init is looking for the mosaic files in the wrong location. The chgres scripts all set `FIX_ORO=${FIX_FV3}/fix_fv3_gmted2010` instead of `${FIX_FV3}/orog`.

Good catch.

GeorgeGayno-NOAA commented:
The `snow2mdl`, `chgres_cube` and `global_cycle` regression tests have old 'fix' paths.

GeorgeGayno-NOAA reviewed (changes requested):
And two paths in this script need to be updated - ./util/sfc_climo_gen/run.wcoss2.sh.

WalterKolczynski-NOAA committed:
More updates for fix reorg

Some additional scripts needed to be updated for the fix reorganization.

A few files still contain paths remincent of the old paths, either to places
the data is hosted on the web (FTP or AWS), or hard-coded locations not under
global-workflow control.

Refs: #688

GeorgeGayno-NOAA commented on ush/chgres_cube.sh:
```diff
@@ -33,9 +33,9 @@ NWROOT=${NWROOT:-/nw${envir}}
-FIXsfc=${FIXsfc:-$FIXfv3/fix_sfc}
-FIXam=${FIXam:-$FIXufs/fix_am}
+FIXfv3=${FIXfv3:-$FIXufs/orog/C${CRES}}
+FIXsfc=${FIXsfc:-$FIXfv3/sfc}
```
This is a different FIXsfc directory. It needs to be reverted to $FIXfv3/fix_sfc.

GeorgeGayno-NOAA replied:
@WalterKolczynski-NOAA This is causing the chgres_cube regression tests to fail.

WalterKolczynski-NOAA commented:
@GeorgeGayno-NOAA I also have a couple local changes I made to do my testing, I don't know if you wanted me to add them to this PR:

1. I created a `cmake/mpiexec.wcoss2` file and edited `cmake/LibMPI.cmake` to submit jobs on cactus.
2. I increased the `pull_data` timeout of `tests/CmakeLists.txt` because the data download would fail with the shorter time.

Should I:
- Add them to this PR?
- Make a separate PR(s)?
- Just point you towards them?
- Ignore them?

GeorgeGayno-NOAA commented:
> @GeorgeGayno-NOAA I also have a couple local changes I made to do my testing, I don't know if you wanted me to add them to this PR:
> 
> 1. I created a `cmake/mpiexec.wcoss2` file and edited `cmake/LibMPI.cmake` to submit jobs on cactus.
> 2. I increased the `pull_data` timeout of `tests/CmakeLists.txt` because the data download would fail with the shorter time.
> 
> Should I:
> 
> * Add them to this PR?
> * Make a separate PR(s)?
> * Just point you towards them?
> * Ignore them?

Let's add them under a separate issue/PR. I tried to create a mpiexec.wcoss2 file one afternoon. I could not get it to work. Thanks for doing this.

WalterKolczynski-NOAA commented:
> > @GeorgeGayno-NOAA I also have a couple local changes I made to do my testing, I don't know if you wanted me to add them to this PR:
> > 
> > 1. I created a `cmake/mpiexec.wcoss2` file and edited `cmake/LibMPI.cmake` to submit jobs on cactus.
> > 2. I increased the `pull_data` timeout of `tests/CmakeLists.txt` because the data download would fail with the shorter time.
> > 
> > Should I:
> > 
> > * Add them to this PR?
> > * Make a separate PR(s)?
> > * Just point you towards them?
> > * Ignore them?
> 
> Let's add them under a separate issue/PR. I tried to create a mpiexec.wcoss2 file one afternoon. I could not get it to work. Thanks for doing this.

Opened two new issue (#690 and #691). I already have the fixes, I just need to sort them into PRs.

DavidHuber-NOAA commented:
@WalterKolczynski-NOAA The chgres_cube baselines store the surface fix files in `${HOMEreg}/fix/C${RES}/fix_sfc`, which are now inconsistent with the fix structure.  When the chgres_cube tests run, they are now looking in `sfc` instead of `fix_sfc`, which does not exist in the baseline input data.  This could be remedied by adding

`export FIXsfc=${FIXfv3}/fix_sfc`

to each of the chgres_cube test scripts (e.g. c96.gfs.sigio.sh).  Do you think this should happen or that the baseline surface fix directories should be renamed on each HPC?

WalterKolczynski-NOAA commented:
@DavidHuber-NOAA I would recommend doing what we did when we made this change to fix: move the baseline to the new locations and temporarily set up symlinks pointing the old locations to the new, so older versions don't break.

WalterKolczynski-NOAA commented:
Also, are these different tests than I get when I run `make test`? If so, I wasn't able to figure out how to run them, otherwise I would have.

DavidHuber-NOAA commented:
@WalterKolczynski-NOAA OK, that sounds reasonable to me regarding the links, though I would defer to @GeorgeGayno-NOAA.  The regression tests are run via [rt.sh](https://github.com/ufs-community/UFS_UTILS/blob/develop/reg_tests/rt.sh).  That script needs to be modified first at the top by specifying a [working directory, account, etc](https://github.com/ufs-community/UFS_UTILS/blob/c33ea5242e7b2fa1c4199a8ff5b88523e371ba01/reg_tests/rt.sh#L5-L11).  Running `make test` runs the unit tests and not the regression tests.

GeorgeGayno-NOAA commented:
> Also, are these different tests than I get when I run `make test`? If so, I wasn't able to figure out how to run them, otherwise I would have.

Only the chgres_cube regression tests are failing. If you make the fix to ./ush/chgres_cube.sh (see my comment) I can run them for you.

WalterKolczynski-NOAA commented:
> > Also, are these different tests than I get when I run `make test`? If so, I wasn't able to figure out how to run them, otherwise I would have.
> 
> Only the chgres_cube regression tests are failing. If you make the fix to ./ush/chgres_cube.sh (see my comment) I can run them for you.

Which comment? I think I've already made the updates from [this comment](#issuecomment-1234575974). I think the remaining issues lie in the baseline location, which is what David and I were talking about yesterday.

WalterKolczynski-NOAA commented:
Is there anything you are waiting on from me that is holding this up?

GeorgeGayno-NOAA commented:
> Is there anything you are waiting on from me that is holding this up?

Merge in the latest updates from 'develop' to your branch. And did you address the problem with the chgres_cube regression test script? See my review.

GeorgeGayno-NOAA commented:
I can run the chgres test for you after you update the script.

GeorgeGayno-NOAA commented on fix/link_fixdirs.sh:
```diff
@@ -35,23 +35,28 @@ pwd=$(pwd -P)
-    FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix_NEW"
+    FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix"
 elif [ $machine = "jet" ]; then
     FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix_NEW"
```
@WalterKolczynski-NOAA the directory on Jet does not exist. Do you mean 'fix' instead of 'fix_NEW'?

WalterKolczynski-NOAA commented:
I updated the branch. I think the issue your talking about are resolve, but see my earlier comment asking which comment you are referring to and the discussion with David about whether the baseline fix files need to be changed to match.

GeorgeGayno-NOAA commented:
@WalterKolczynski-NOAA Please make this change:

```
diff --git a/ush/chgres_cube.sh b/ush/chgres_cube.sh
index 027b1e37..40d91f54 100755
--- a/ush/chgres_cube.sh
+++ b/ush/chgres_cube.sh
@@ -34,7 +34,7 @@ HOMEufs=${HOMEufs:-${NWROOT}/ufs_util.${ufs_ver}}
 EXECufs=${EXECufs:-$HOMEufs/exec}
 FIXufs=${FIXufs:-$HOMEufs/fix}
 FIXfv3=${FIXfv3:-$FIXufs/orog/C${CRES}}
-FIXsfc=${FIXsfc:-$FIXfv3/sfc}
+FIXsfc=${FIXsfc:-$FIXfv3/fix_sfc}
 FIXam=${FIXam:-$FIXufs/am}
```

GeorgeGayno-NOAA commented:
@WalterKolczynski-NOAA And the jet directories need to be updated:

```
/lfs4/HFIP/emcda/George.Gayno/ufs_utils.git/UFS_UTILS.walter/fix $ git diff link_fixdirs.sh
diff --git a/fix/link_fixdirs.sh b/fix/link_fixdirs.sh
index ee19afa2..a236de8e 100755
--- a/fix/link_fixdirs.sh
+++ b/fix/link_fixdirs.sh
@@ -37,7 +37,7 @@ pwd=$(pwd -P)
 if [ $machine = "hera" ]; then
     FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix"
 elif [ $machine = "jet" ]; then
-    FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix_NEW"
+    FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix"
 elif [ $machine = "orion" ]; then
     FIX_DIR="/work/noaa/global/glopara/fix"
 elif [ $machine = "wcoss2" ]; then
```

WalterKolczynski-NOAA committed:
Fix sfc fix for chgres_cube and fix path on Jet

Refs: #687

GeorgeGayno-NOAA reviewed (approved):
All regression tests now passing on all machines. Thanks, @WalterKolczynski-NOAA
